test/spec/Feature/OpenApi/DisabledOpenApiSpec.hs

```diff
@@ -3,14 +3,18 @@ module Feature.OpenApi.DisabledOpenApiSpec where
 import Network.HTTP.Types
 import Network.Wai        (Application)
 
-import Test.Hspec     hiding (pendingWith)
+import Test.Hspec
 import Test.Hspec.Wai
+import Test.Hspec.Wai.JSON
 
 import Protolude
 
 spec :: SpecWith ((), Application)
 spec =
   describe "Disabled OpenApi" $ do
-    it "responds with 404" $
+    it "responds with 400" $
       request methodGet "/"
-        [("Accept","application/openapi+json")] "" `shouldRespondWith` 404
+        [("Accept","application/openapi+json")] ""
+        `shouldRespondWith`
+        [json| {"code":"PGRST126","details":null,"hint":"Enable configuration `openapi-mode`","message":"Open API mode is disabled but API root path is accessed"} |]
```

Comment:
Since this is a message for an API user, we shouldn't reveal server details.

```suggestion
        [json| {"code":"PGRST126","details":null,"hint": null,"message":"Open API mode is disabled"} |]
```

However the error code section in the docs can mention `openapi-mode`.